agent/config/builder.go

```diff
@@ -633,6 +633,22 @@ func (b *Builder) Build() (rt RuntimeConfig, err error) {
 		verifyOutgoing = true
 	}
 
+	var configEntries []structs.ConfigEntry
+
+	if len(c.Config.ProxyDefaults) > 0 {
+		for name, config := range c.Config.ProxyDefaults {
```

Comment:
Do we need to support a list of entries for proxy defaults if we only support setting the global one currentlyt?

Reply:
I initially thought the same but actually I guess this is more flexible in the future and means the config file nesting maps the same way for all config entry types (e.g. CLI will expose managed proxy_defaults with name "global" so it kinda makes sense to do the same here.

Reply:
Originally I structure the non-runtime config to have an explicit "global" struct under the "proxy_defaults" struct. It made it harder to detect existence or not (although I could have probably made the data a pointer to a map[string]interface{}).

Allowing the config to accept a bunch but then handling validation later made things a bit cleaner IMO and would allow us flexibility in the future if we wanted to add an other proxy default.